File: frontend/src/types/agent.ts
// === Agent Module Types ===

import type { PipelineNode } from './index';

/** Agent info from GET /agents/available */
export interface AgentInfo {
  name: string;
  display_name: string;
  icon: string;
  description: string;
  modes: string[];
  available: boolean;
  version: string | null;
  provider: string | null;
  model: string | null;
  install_hint: string | null;
  tools: string[];
  mcp_servers: string[];
}

/** Session record from backend */
export interface AgentSession {
  id: string;
  agent_name: string;
  native_session_id: string | null;
  title: string;
  mode: string;
  source: string;
  created_at: string;
  updated_at: string;
}

/** WebSocket client → server message */
export interface AgentChatMessage {
  type: 'message';
  content: string;
  mode: string;
  context?: AgentContextData | null;
}

export interface AgentContextData {
  type: 'skill' | 'pipeline' | 'free';
  skill?: Record<string, unknown>;
  pipeline?: Record<string, unknown>;
  selected_node?: string;
  error_result?: Record<string, unknown>;
  attachments?: Attachment[];
}

export interface Attachment {
  type: 'skill' | 'pipeline' | 'text';
  label: string;
  content: string;
}

/** Server → client event via WebSocket */
export interface AgentEvent {
  type: 'text' | 'code' | 'error' | 'session_init' | 'done';
  content: string;
  metadata: Record<string, unknown>;
}

/** Internal chat message for UI rendering */
export interface ChatMessage {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  timestamp: number;
}

/** Pipeline action suggested by agent */
export type PipelineAction =
  | { action: 'add_node'; skill_name: string; position: number; config_overrides?: Record<string, unknown> }
  | { action: 'remove_node'; node_id: string }
  | { action: 'update_node'; node_id: string; changes: Partial<PipelineNode> }
  | { action: 'update_config'; node_id: string; config_overrides: Record<string, unknown> };

File: backend/app/schemas/agent.py
"""Pydantic schemas for the Agent module."""

from datetime import datetime

from pydantic import BaseModel, Field


class AgentInfoResponse(BaseModel):
    name: str
    display_name: str
    icon: str
    description: str
    modes: list[str]
    available: bool
    version: str | None = None
    provider: str | None = None
    model: str | None = None
    install_hint: str | None = None
    tools: list[str] = []
    mcp_servers: list[str] = []


class CreateSessionRequest(BaseModel):
    agent_name: str
    source: str = "playground"
    mode: str = "code"


class AgentSessionResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    agent_name: str
    native_session_id: str | None
    title: str
    mode: str
    source: str
    created_at: datetime
    updated_at: datetime


class AgentChatMessage(BaseModel):
    """Schema for WebSocket client messages."""

    type: str = "message"
    content: str
    mode: str = "code"
    context: dict | None = None


class AgentEventResponse(BaseModel):
    type: str
    content: str = ""
    metadata: dict = Field(default_factory=dict)
